crates/re_edit_ui/src/lib.rs

```diff
@@ -34,46 +33,55 @@ use re_viewer_context::gpu_bridge::colormap_dropdown_button_ui;
+    registry.add_singleline_edit_or_view::<DepthMeter>(edit_f32_zero_to_max_float_raw);
+    registry.add_singleline_edit_or_view::<MarkerSize>(edit_f32_zero_to_max_float_raw);
+    registry.add_singleline_edit_or_view::<StrokeWidth>(edit_f32_zero_to_max_float_raw);
 
-    registry.add_singleline_editor_ui::<Opacity>(edit_f32_zero_to_one);
+    registry.add_singleline_edit_or_view::<Opacity>(edit_f32_zero_to_one);
 
-    registry.add_singleline_editor_ui::<Visible>(edit_bool_raw);
-    registry.add_singleline_editor_ui::<LockRangeDuringZoom>(edit_bool);
+    registry.add_singleline_edit_or_view::<Visible>(edit_bool_raw);
+    registry.add_singleline_edit_or_view::<LockRangeDuringZoom>(edit_bool);
 
-    registry.add_singleline_editor_ui::<Text>(edit_singleline_string);
     registry.add_display_ui(Text::name(), Box::new(display_text_ui));
-    registry.add_singleline_editor_ui::<Name>(edit_singleline_string);
+    registry.add_singleline_edit_or_view::<Text>(edit_singleline_string);
+    registry.add_multiline_edit_or_view::<Text>(edit_multiline_string);
     registry.add_display_ui(Name::name(), Box::new(display_name_ui));
+    registry.add_singleline_edit_or_view::<Name>(edit_singleline_string);
+    registry.add_multiline_edit_or_view::<Name>(edit_multiline_string);
 
-    registry.add_singleline_editor_ui(|_ctx, ui, value| edit_enum::<BackgroundKind>(ui, value));
-    registry.add_singleline_editor_ui(|ctx, ui, value| {
-        response_with_changes_of_inner(colormap_dropdown_button_ui(ctx.render_ctx, ui, value))
+    registry
+        .add_multiline_edit_or_view(|_ctx, ui, value| edit_view_enum::<BackgroundKind>(ui, value));
+    registry.add_multiline_edit_or_view(|ctx, ui, value| {
+        colormap_dropdown_button_ui(ctx.render_ctx, ui, value)
     });
-    registry.add_singleline_editor_ui(|_ctx, ui, value| edit_enum::<Corner2D>(ui, value));
+    registry.add_multiline_edit_or_view(|_ctx, ui, value| edit_view_enum::<Corner2D>(ui, value));
     registry
-        .add_singleline_editor_ui(|_ctx, ui, value| edit_enum::<MagnificationFilter>(ui, value));
-    registry.add_singleline_editor_ui(|_ctx, ui, value| edit_enum::<AggregationPolicy>(ui, value));
-    registry.add_singleline_editor_ui(|_ctx, ui, value| edit_enum::<ViewFit>(ui, value));
+        .add_multiline_edit_or_view(|_ctx, ui, value| edit_view_enum::<BackgroundKind>(ui, value));
+    registry.add_multiline_edit_or_view(|_ctx, ui, value| edit_view_enum::<Colormap>(ui, value));
+    registry.add_multiline_edit_or_view(|_ctx, ui, value| edit_view_enum::<Corner2D>(ui, value));
+    registry.add_multiline_edit_or_view(|_ctx, ui, value| {
+        edit_view_enum::<MagnificationFilter>(ui, value)
+    });
+    registry.add_multiline_edit_or_view(|_ctx, ui, value| {
+        edit_view_enum::<AggregationPolicy>(ui, value)
+    });
+    registry.add_multiline_edit_or_view(|_ctx, ui, value| edit_view_enum::<ViewFit>(ui, value));
```

Comment:
we don't. support multiline editors on defaults and visualizer ui yet. This ports a lot of things to multi-line only, meaning lots of things can no longer be edited 

<img width="239" alt="image" src="https://github.com/rerun-io/rerun/assets/1220815/4b34c67c-ff49-424b-8227-a859a8a4ef43">

Reply:
actually, why are those enums all marked as multiline editors. Oversight or a deeper reason I'm missing?

Reply:
It's an oversight; these should all be singleline